generator/lib/python.ml

```diff
@@ -46,7 +46,14 @@ let rec typecheck env ty v =
   | Struct (hd, tl) ->
     let member (name, ty, descr) =
       typecheck env ty (sprintf "%s['%s']" v name) in
-    List.concat (List.map member (hd :: tl))
+    let member_maybe_opt (name, ty, descr) =
+      match ty with
+      | Option _ -> [
+        Line (sprintf "if '%s' in %s:" name v);
+        Block (member (name, ty, descr))
+      ]
+      | _ -> member (name, ty, descr) in
+    List.concat (List.map member_maybe_opt (hd :: tl))
```

Comment:
So basically what we do here is that we do not typecheck the fields in the struct if they are missing, which would result in a runtime error.

Reply:
Only for optional fields though, i.e. if a struct field is optional we treat the absence of that field as if it was None.

Reply:
This is used by the new 'sr' field  in probe result which is optional, and `xapi-storage-plugins` simply  doesn't send it when its empty, but on the ocaml side we want to treat it as none (in fact the rpc code already treats it that way IIUC).